libr/bin/format/pe/pe.c

```diff
@@ -2584,11 +2584,13 @@ struct PE_(r_bin_pe_obj_t)* PE_(r_bin_pe_new)(const char* file) {
 }
 
 struct PE_(r_bin_pe_obj_t)* PE_(r_bin_pe_new_buf)(struct r_buf_t *buf) {
+	RIOBind *iob = (RIOBind *)(buf->iob);
 	struct PE_(r_bin_pe_obj_t) *bin = R_NEW0 (struct PE_(r_bin_pe_obj_t));
 	if (!bin) return NULL;
 	bin->kv = sdb_new0 ();
 	bin->b = r_buf_new ();
-	bin->size = buf->length;
+	bin->size = (iob && iob->io) ? r_io_size (iob->io) : buf->length;
```

Comment:
what about extend `r_buf_size` api to handle this. We should use more api and avoid RIOBind to operate, doing everything through r_buf